### PR TITLE
fix(select): fixed click events which were not working

### PR DESCRIPTION
**Pull request by jhuntoo**

### CODE REVIEW

File: components/select/off-click.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import {Directive, Host, HostListener, Input} from 'angular2/core';
```
Comment (jhuntoo) [inline components/select/off-click.ts:1]: New Directive which will fire an event when a click occurs outside of the element

File: components/select/select.ts [hunk 2]
```diff
@@ -239,41 +243,22 @@ export class Select {
   }
 
   ngOnInit() {
-    this.behavior = this.itemObjects[0].hasChildren() ?
```
Comment (jhuntoo) [inline components/select/select.ts:242]: This line was breaking when 0 items were bound.

File: components/select/select.ts [hunk 3]
```diff
@@ -239,41 +243,22 @@ export class Select {
...
 
-  ngOnDestroy() {
-    document.removeEventListener('click', this.offSideClickHandler);
-    this.offSideClickHandler = null;
+  clickedOutside()  {
+    this.inputMode = false;
+    this.optionsOpened = false
   }
 
-  private getOffSideClickHandler(context:any) {
```
Comment (jhuntoo) [inline components/select/select.ts:258]: Simplified with new directive and the above event handler.

File: components/select/select.spec.ts [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+  ComponentFixture,
+  expect
+
+} from 'angular2/testing';
+
+import {Component, provide} from 'angular2/core';
+
+import {Select} from './select';
+
+describe('Select', () => {
```
Comment (jhuntoo) [inline components/select/select.spec.ts:17]: A bit of repetition here - can be DRY'd later.

File: tslint.json [hunk 5]
```diff
@@ -1,56 +1,76 @@
 {
```
Comment (jhuntoo) [inline tslint.json:1]: this can be replace with the valor-software standard later on

**Comment by valorkin** [comment]
Really good but can merge it
Can you rebase it on top of dev branch

**Comment by valorkin** [comment]
I really like what you have done here
But can we split it in several PRs?
- tests runner first

**Comment by valorkin** [comment]
oh, it was so long time ago :)

**Pull request closed without merging**